pkg/cacheutil/memcached_client.go

```diff
@@ -38,8 +38,10 @@ const (
 )
 
 var (
-	errMemcachedAsyncBufferFull = errors.New("the async buffer is full")
-	errMemcachedConfigNoAddrs   = errors.New("no memcached addresses provided")
+	errMemcachedAsyncBufferFull                = errors.New("the async buffer is full")
+	errMemcachedConfigNoAddrs                  = errors.New("no memcached addresses provided")
+	errMemcachedDNSUpdateIntervalNotPositive   = errors.New("dns_provider_update_interval must be positive")
+	errMemcachedMaxAsyncConcurrencyNotPositive = errors.New("max_async_concurrency must be positive")
```

Comment:
Would you mind not mentioning the config name in the errors, but make it more generic, so that we don't have mismatching issues in Cortex? Cortex config differs.

```suggestion
	errMemcachedDNSUpdateIntervalNotPositive   = errors.New("DNS provider update interval must be positive")
	errMemcachedMaxAsyncConcurrencyNotPositive = errors.New("max async concurrency must be positive")
```

Reply:
Great suggestion! Make sense to me. Thank you!